### DISCUSSION TITLE
testing rtk-query

### DISCUSSION THREAD

**Discussion by ghost** [reduxjs/redux-toolkit#1262] [Q&A] [answered]
What is the recommened way to test `rtk-query`? I want to ensure that matchers for `apiSlice.endpoints.get.matchFulfilled` work.

```
it('example', () => {
      fetchMock.mockResponseOnce(JSON.stringify(payload));

      store.dispatch(apiSlice.endpoints.get {HERE});
});
```

**Comment by markerikson** [comment, marked as answer]
My first reaction is that you shouldn't have to test those at all - you're testing the actual implementation details of RTK Query itself, and we have tests for that already :)

**Reply by ghost** [reply]
Hey thanks for the quick response! Maybe I should have phrased my question better. 

I have other slices that update their store when `apiSlice.endpoints.get.matchFulfilled` is matched. I want to test that their slices are updated when that action is fired. I would like to write this test more to ensure developers in the future don't remove certin logic in those matchers. Maybe there is a better way to ensure this! Happy for suggestions if you have any :)

**Reply by markerikson** [reply]
Ah, gotcha.

In that case, you could potentially dispatch the relevant RTKQ thunks manually in the test:

https://redux-toolkit.js.org/rtk-query/usage/usage-without-react-hooks

assuming you've mocked out the API calls.

I can't remember if the individual action creators are exposed or not - off the top of my head I don't think they are, but I could be wrong.

**Reply by phryneas** [reply]
They are not exposed and I would also not try to fake them - at least make a "real request" against your fake api (we use msw) by dispatching an `.invoke` call or you won't be sure that you are actually reacting to something RTKQ really dispatches.

**Reply by ghost** [reply]
ok great, wasn't sure what the right direction was. thank you for the guidance @markerikson and @phryneas!

**Reply by arikark** [reply]
What would the approach to dispatching the relevant RTKQ thunks manually in the test if the endpoint is 'injected' with code splitting?

**Reply by phryneas** [reply]
You should export the endpoint/hook/whatever your component uses from the injecting file and also import it from that. That one you make 100% sure that your injecting file is always loaded when you access something from it.
